ballerina/annotations.bal

```diff
@@ -14,14 +14,20 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import ballerina/http;
+
 # Provides a set of configurations for the GraphQL service.
 #
 # + maxQueryDepth - The maximum depth allowed for a query
 # + auth - Listener authenticaton configurations
+# + contextInit - Function to initialize the context. If not provided, an empty context will be created
 public type GraphqlServiceConfig record {|
     int maxQueryDepth?;
     ListenerAuthConfig[] auth?;
+    ContextInit contextInit?;
```

Comment:
What if we assign a default value for this ? Something like

```ballerina
isolated function initDefaultContext(http:Request request, http:RequestContext requestContext) returns graphql:Context|error {
    return graphql:Context();
}
```

That way it is explicit that we create a default one if nothing is given.

Reply:
We already have a function for this:
https://github.com/ballerina-platform/module-ballerina-graphql/blob/df9a8fae41edc30c81f6616044378b866008a943/ballerina/context.bal#L68

Reply:
@ThisaruGuruge @shafreenAnfar What do you think about the field name `contextInit`? I prefer this to be as `Context context`, and passing a function pointer which creates a `Context` for me. But I understand that it is not possible with Ballerina way of handling types. So, what do you think of having this as `ContextInit context`?

Reply:
At first, I went through the same thought process. But then went with `contextInit` because this isn't actually a `context`, but a function that initializes a context.
IMHO, this makes it easy for the developers to understand what it is. They have to provide a `ContextInit` (which is a function pointer), not a `Context` (which is an object).

Reply:
Yeah. Agree with the clarification.